frontend/src/pages/SchedulePage.tsx

```diff
@@ -12,49 +12,68 @@ import AdapterDateFns from "@mui/lab/AdapterDateFns";
+import {DepositDTO} from "../models/Deposit";
 
 interface SelectInput {
     selectCategory: { value: string }
 }
 
 export default function SchedulePage() {
     const config = useContext(AuthContext).config
-    const [series, setSeries] = useState<Series[]>([])
     const [scheduledDate, setScheduledDate] = useState<number>(1)
     const [description, setDescription] = useState<string>("")
+    const [typeName, setTypeName] = useState<string>("Payment")
     const [amount, setAmount] = useState<number>(1)
+
+    const [paymentSeries, setPaymentSeries] = useState<PaymentSeries[]>([])
+    const [depositSeries, setDepositSeries] = useState<DepositSeries[]>([])
     const [categories, setCategories] = useState<Category[]>([])
     const [rangeValue, setRangeValue] = useState<DateRange<Date>>([null, null])
     const seriesController: ISeriesController = useMemo(() => SeriesController(config), [config])
     const categoryController: ICategoryController = useMemo(() => CategoryController(config), [config])
 
     useEffect(() => {
-        seriesController.getSeries().then(setSeries)
+        seriesController.getPaymentSeries().then(setPaymentSeries)
+        seriesController.getDepositSeries().then(setDepositSeries)
         categoryController.getCategories().then(setCategories)
     }, [seriesController, categoryController])
 
     const addSeries = (event: FormEvent<HTMLFormElement>) => {
         event.preventDefault()
-        const form = event.currentTarget
-        const formElements = form.elements as typeof form.elements & SelectInput
-        const categoryID: string = formElements.selectCategory.value
+        if (typeName === "Payment") {
+            const form = event.currentTarget
+            const formElements = form.elements as typeof form.elements & SelectInput
+            const categoryID: string = formElements.selectCategory.value
 
-        if (!categoryID || !categoryID.length) return
+            if (!categoryID || !categoryID.length) return
 
-        const payment: PaymentDTO = {description, amount, categoryID}
-        const seriesObj: Series = {scheduledDate, payment, startDate: rangeValue[0], endDate: rangeValue[1]}
-        seriesController.addSeries(seriesObj).then(setSeries)
+            const payment: PaymentDTO = {description, amount, categoryID}
+            const seriesObj: PaymentSeries = {scheduledDate, payment, startDate: rangeValue[0], endDate: rangeValue[1]}
+            seriesController.addPaymentSeries(seriesObj).then(setPaymentSeries)
+        } else {
+            const deposit: DepositDTO = {description, amount}
+            const seriesObj: DepositSeries = {scheduledDate, deposit, startDate: rangeValue[0], endDate: rangeValue[1]}
+            seriesController.addDepositSeries(seriesObj).then(setDepositSeries)
+        }
     }
 
-    const deleteSeries = (seriesId: string | undefined) => {
+    const deleteSeries = (seriesId: string | undefined, type: string) => {
```

Comment:
```suggestion
    const deleteSeries = (seriesId?: string, type: string) => {
```

Reply:
If you make your id mandatory, (which series doesnt has an id?) you can save some checks